integrated_channels/blackboard/client.py

```diff
@@ -57,17 +59,40 @@ def create_content_metadata(self, serialized_data):
 
         external_id = channel_metadata_item.get('externalId')
 
+        # Retrieve and pop the data not needed for course creation and necessary for content customization
+        enrollment_link_metadata = channel_metadata_item.pop('enrollment_metadata')
+        bb_course_description = enrollment_link_metadata.get('element')
+        course_content_child_data = self.generate_integration_content_data(
+            channel_metadata_item.get('name'),
+            enrollment_link_metadata.get('link'),
+            channel_metadata_item.pop('syllabus_body'),
+            channel_metadata_item.pop('image_url')
+        )
+
         # blackboard does not support all characters in our courseIds so let's gen a hash instead
         course_id_generated = self.generate_blackboard_course_id(external_id)
 
         copy_of_channel_metadata = copy.deepcopy(channel_metadata_item)
         copy_of_channel_metadata['courseId'] = course_id_generated
+        copy_of_channel_metadata['description'] = bb_course_description
 
         LOGGER.info("Creating course with courseId: %s", external_id)
         self._create_session()
         create_url = self.generate_course_create_url()
         response = self._post(create_url, copy_of_channel_metadata)
-        return response.status_code, response.text
+
+        bb_course_id = response.json().get('id')
+
+        LOGGER.info("Creating content page for Blackboard course with course ID={}".format(external_id))
+        bb_content_id = self.create_integration_content_for_course(bb_course_id, course_content_child_data)
+
+        success_body = 'Successfully created Blackboard integration course={bb_course_id} with integration ' \
+                       'content={bb_content_id}'.format(
+                           bb_course_id=bb_course_id,
+                           bb_content_id=bb_content_id,
+                       )
+
+        return HTTPStatus.OK.value, success_body
```

Comment:
I think it may be better to return the actual response code here from the prior post? alternatively, you can just log that response code in the log above? just wondering if we can get codes that are not 200 but still success that will be good to know?

Reply:
Do you want to return the status code from the initial course creation? From the content creation? From the content child creation? I'm ok with taking one of those responses, but the reason why I did the generic `200` was because I felt like there was some ambiguity here. Are you ok with returning the success body?

Reply:
good question. I think that base IntegratedChannelApiClient makes _no_ comments or restrictions about what this method should return if anything, the output is never consumed anywhere. I think we introduced this notion in our code?

if we want the return values as is, I would say let's return the most recent call's status and body. If anything fails we return that status and body at the time of failure

however, we have the option to also not return anything from this method since no one seems to be consuming it and the base class does not require it

the only use of this method seems to be here

https://github.com/edx/edx-enterprise/blob/4b8faec47ae0aa74e5a3f941035e5f35e88a807e/integrated_channels/integrated_channel/transmitters/content_metadata.py#L127

I personally do not like method that return nothing. I usually at least return a boolean success (or in a fluent api return the object instance). But it is what it is right now

Does this work?

Reply:
> I think we introduced this notion in our code?

I think that I introduced this behavior with learner data because we record and store the returned body and status in order to figure out future transmission updates/deletes. Seems like it got transferred over into content metadata somewhere... 

I agree that returning nothing seems like a step in the wrong direction. Let me put up a change and we can see if that satisfies everyone 😄